remotespark/livyclientlib/pandaspysparklivyclient.py

```diff
@@ -20,11 +24,34 @@ def no_records(self, records_text):
         return records_text == "[]"
 
     def get_columns_dataframe(self, columns_text):
-        records = list()
-        columns = eval(columns_text)
-        return pd.DataFrame.from_records(records, columns=columns)
+        return pd.DataFrame.from_records([], columns=self._extract_strings_from_array(columns_text))
+
 
     def get_data_dataframe(self, records_text):
-        json_data = eval(records_text)
-        json_array = "[{}]".format(",".join(json_data))
-        return pd.DataFrame(json.loads(json_array))
+        strings = self._extract_strings_from_array(records_text)
+        try:
+            return pd.DataFrame([json.loads(s) for s in strings])
+        except ValueError:
+            raise DataFrameParseException("Cannot parse object as JSON: '{}'".format(strings))
```

Comment:
This will be logged and it will include customer data. Please check the PR for that.